internal/app/init/main.go

```diff
@@ -195,6 +195,46 @@ func root() (err error) {
 		if _, err = kmsg("[talos]"); err != nil {
 			return fmt.Errorf("failed to setup logging to /dev/kmsg: %v", err)
 		}
+
+		log.Println("mounting overlay for /etc/kubernetes")
```

Comment:
This is the new way of having writable paths on top of the `rootfs` without the need for symlinks and/or extra partitions.